src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -46,8 +46,14 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
         /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
         public async Task Invoke(HttpContext context)
         {
-            var waitInQueueTask = _requestQueue.EnterQueue();
+            if (_requestQueue.QueueLengthExceeded)
+            {
+                context.Response.StatusCode = 503;
+                context.Abort();
```

Comment:
Abort not required

Reply:
The 503 seems a bit dramatic.

Reply:
The 503 seems a bit dramatic.

Reply:
@DAllanCarr how so? What else would we respond with?

Reply:
This is not rate limiting I assume. I was thinking 503 maybe misinterpreted.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
"The server is not ready to handle the request. Common causes are a server that is down for maintenance or that is overloaded."

Reply:
@DAllanCarr no, this isn't rate limiting, though it's something we have on the radar for this project eventually. Right now the scope is just about having a single queue for all requests coming in to the server to help manage resources better in certain environments.